### PR TITLE
Read `idmap` source from `golang` code

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

I added some patches to first read `idmap` from `golang` code, so we are able to `open_tree` and set the attribute of a custom path.
Some particular points on specific commits:

1. [fixup! libcontainer: Add UID/GID mappings to mounts](https://github.com/kinvolk/runc/pull/61/commits/52fe299928a06c6675f1380ff721ce253a117578) this fixes the original commit so `runc` can read the UID and GID mappings from `config.json`.
2. [Use IsBind rather than bit testing.](https://github.com/kinvolk/runc/pull/61/commits/fb8d3e5a395a5be8064c72ac96bc30f0143d8853) is a small patch which can be upstreamed right now.
3. [wip: libcontainer: Send mount id mapped sources from golang.](https://github.com/kinvolk/runc/pull/61/commits/f3176b8f46135e949f620080ce5a0d4913ae3e3f) sends the source paths from `golang` to `C`.
4. [fixup! WIP: Mount a hardcoded idmap mount](https://github.com/kinvolk/runc/commit/414aa3c6cca87f9c42542c6eae1a2ed3f08c0fb5) runs `make cfmt` on previous code so we are the same page.
5. [wip: open_tree and set_attr of mount sources received from golang.](https://github.com/kinvolk/runc/pull/61/commits/e183d4e2442bfd3fb6b6536c47da4b335843454e)](https://github.com/kinvolk/runc/pull/61/commits/e183d4e2442bfd3fb6b6536c47da4b335843454e) actually modifies your code to `open_tree` and `set_attr` the sources path.

@rata :D

Best regards and thank you in advance.

### CODE REVIEW

File: libcontainer/nsenter/nsexec.c [hunk 1]
```diff
@@ -37,27 +37,27 @@
 
 /* mount_setattr() */
 #ifndef MOUNT_ATTR_IDMAP
-#define MOUNT_ATTR_IDMAP 0x00100000
+#  define MOUNT_ATTR_IDMAP 0x00100000
```
Comment (rata) [inline libcontainer/nsenter/nsexec.c]: Is the linter being silly? The space after `#` is not any style I'm familiar with... It smells as it thinks `#` is used for comments and it is indenting it as if it were comments

Comment (eiffel-fl) [reply libcontainer/nsenter/nsexec.c]: Highly possible, I will update this commit to not format this part.

File: libcontainer/specconv/spec_linux.go [hunk 2]
```diff
@@ -519,6 +538,9 @@ func createLibcontainerMount(cwd string, m specs.Mount) (*configs.Mount, error)
 		}
 	}
 
+	mnt.UIDMaps = toIDMapArray(m.UIDMappings)
+	mnt.GIDMaps = toIDMapArray(m.GIDMappings)
```
Comment (rata) [inline libcontainer/specconv/spec_linux.go:542]: Oh, I missed this, cool!

**Comment by rata** [comment]
This is the branch I used for the PR upstream. Will close the PR and merge into `rata/idmap-wip`

**Comment by rata** [comment]
@eiffel-fl btw, please open a PR upstream for commit number 2!

**Pull request closed without merging**